examples/flower-secure-aggregation/secaggexample/server_app.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Tuple
+
+from secaggexample.task import IS_DEMO, Net, get_weights
+from secaggexample.workflow_with_log import SecAggPlusWorkflowWithLogs
+
+from flwr.common import Context, Metrics, ndarrays_to_parameters
+from flwr.server import Driver, LegacyContext, ServerApp, ServerConfig
+from flwr.server.strategy import FedAvg
+from flwr.server.workflow import DefaultWorkflow, SecAggPlusWorkflow
+
+
+# Define metric aggregation function
+def weighted_average(metrics: List[Tuple[int, Metrics]]) -> Metrics:
+    # Multiply accuracy of each client by number of examples used
+    accuracies = [num_examples * m["accuracy"] for num_examples, m in metrics]
+    examples = [num_examples for num_examples, _ in metrics]
+
+    # Aggregate and return custom metric (weighted average)
+    return {"accuracy": sum(accuracies) / sum(examples)}
+
+
+ndarrays = get_weights(Net())
+parameters = ndarrays_to_parameters(ndarrays)
```

Comment:
Let's better move it inside `@app.main()` 🙏

Reply:
Sure!